app/controllers/visual_groups_controller.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+class VisualGroupsController < ApplicationController
+  before_action :login_required
+
+  # GET /visual_groups or /visual_groups.json
+  def index
+    @visual_groups = VisualGroup.joins(
+      "LEFT JOIN names ON visual_groups.group_name_id = names.id"
+    ).order(
+      "names.text_name"
+    )
+  end
```

Comment:
@mo-nathan 
Joe, Jason and I have done a lot of work to rephrase SQL literals with the equivalent AR — our goal for test coverage and database portability is no SQL strings. I believe this will be just as efficient with
```ruby
  def index
    @visual_groups = VisualGroup.joins(:names).order(:text_name)
  end
```

Reply:
Actually that doesn't work.  I tried something like that initially but the problem is that generates:

`SELECT visual_groups.* FROM visual_groups INNER JOIN names ON names.visual_group_id = visual_groups.id ORDER BY text_name ASC`

rather than the intended

`SELECT visual_groups.* FROM visual_groups LEFT JOIN names ON visual_groups.group_name_id = names.id ORDER BY text_name ASC`

The first query returns a bunch of duplicate VisualGroups.  However, I did adopt the `.order(:text_name)`.  I wasn't sure that was going to work, but it does.

Reply:
Agh sorry Nathan... It's `VisualGroup.left_outer_joins(:names).order(:text_name)`

https://guides.rubyonrails.org/v6.1.6/active_record_querying.html#left-outer-joins

Reply:
Actually that's even worse.  That generates:
`SELECT visual_groups.* FROM visual_groups LEFT OUTER JOIN names ON names.visual_group_id = visual_groups.id ORDER BY text_name ASC`
which generates even more rows due to the outer join.  The point is that we want to be joining on `visual_groups.group_name_id` not on `names.visual_group_id`.  VisualGroups have only one `group_name_id`, but the relationship from Name to VisualGroup is a many to one relationship.  My code will generate exactly one row for each VisualGroup.  You code generates one row for each Name that has a VisualGroup (and the outer join case throws in a couple more because of the NULL case).

Reply:
Decided to not use Names directly with VisualGroups making this kind of moot.